templates/libvirt/xen.xml

```diff
@@ -23,7 +23,12 @@
         {% block os %}
             {% if vm.hvm %}
                 <type arch="x86_64" machine="xenfv">hvm</type>
-                <loader>hvmloader</loader>
+                <!--
+                     For the libxl backend libvirt switches between OVMF (EFI)
+                     and SeaBIOS based on the loader type. This has nothing to
+                     do with the hvmloader binary.
+                -->
+                <loader type="{{ "rom" if vm.features.check_with_template('legacy-bios', False) else "pflash" }}">hvmloader</loader>
```

Comment:
Since templates by default do not have EFI bootloader installed, the default should be legacy-bios.